corehq/apps/app_manager/models.py

```diff
@@ -1983,6 +2005,19 @@ def wrap(cls, data):
 
         return super(AdvancedForm, cls).wrap(data)
 
+    def _pre_delete_hook(self):
+        try:
+            self.disable_schedule()
+        except (ScheduleError, TypeError, AttributeError):
```

Comment:
Yeah, I should probably add a log here. These exceptions are only thrown if something has gone out of whack (e.g. the form thinks it is in a phase, but the phase disagrees). `AttributeError` gets thrown if the form was created before this PR, i.e. if it doesn't have a `schedule` property. 
Basically, I didn't think that anything bad was happening if these exceptions are raised which is why I didn't handle them past just skipping over it.